src/pages/api/auth/[...auth0].ts

```diff
@@ -20,6 +45,39 @@ const handler: NextApiHandler = handleAuth({
       ...rest,
     });
   },
+  callback: async (req, res) => {
+    return handleCallback(req, res, {
+      afterCallback: async (callbackReq, callbackRes, session) => {
+        const { sub: id, email, name, picture, updatedAt } = session.user;
+
+        assertString(id, "id");
+        assertString(email, "email");
+        assertString(name, "name");
+        assertOptionalString(picture, "picture");
+        assertOptionalString(updatedAt, "updatedAt");
```

Comment:
`session.user` is defined is this callback, but all its keys are `any`, according to Auth0 typings. We could do typecasting here, but that would not be 100% safe. I decided to add [assertion functions](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions) to throw proper errors in runtime and to narrow the types at the same time.